internal/cloud/realm/client_test.go

```diff
@@ -0,0 +1,39 @@
+package realm_test
+
+import (
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	u "github.com/10gen/realm-cli/internal/utils/test"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+)
+
+func TestRealmClient(t *testing.T) {
+	u.SkipUnlessRealmServerRunning(t)
+
+	t.Run("Should work correctly execute request with valid credentials", func(t *testing.T) {
```

Comment:
this test feels similar to the one defined in `TestRealmAuthProfile`.  in addition to potentially changing the implementation of `c.refreshAuth` to leverage `c.do`,  I think we should probably move that function defintion to the `auth.go` file in this package.  thoughts?